src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -75,6 +96,14 @@ public void addPerson(Person p) {
         persons.add(p);
     }
 
+    /**
+     * Adds a listing to the address book.
+     * The listing must not already exist in the address book.
+     */
+    public void addListing(Listing l) {
+        listings.add(l);
```

Comment:
Would be good to add the requreNonNull() check here as well.

Reply:
Resolved